paymentsheet/src/main/java/com/stripe/android/customersheet/data/CachedCustomerEphemeralKey.kt

```diff
@@ -0,0 +1,36 @@
+package com.stripe.android.customersheet.data
+
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.minutes
+
+internal sealed interface CachedCustomerEphemeralKey {
+    fun shouldRefresh(currentTimeInMillis: Long): Boolean
+
+    data class Available(
+        val customerId: String,
+        val ephemeralKey: String,
+        private val expiresAt: Int,
+    ) : CachedCustomerEphemeralKey {
+        override fun shouldRefresh(currentTimeInMillis: Long): Boolean {
+            val remainingTime = expiresAt - currentTimeInMillis.milliseconds.inWholeSeconds
+            return remainingTime <= 5.minutes.inWholeSeconds
+        }
+    }
+
+    data object None : CachedCustomerEphemeralKey {
+        override fun shouldRefresh(currentTimeInMillis: Long): Boolean {
+            return true
+        }
+    }
+}
+
+internal fun <T> Result<CachedCustomerEphemeralKey>.mapWithEphemeralKeyCatching(
```

Comment:
I _think_ that you wouldn't need this function at all if you had changed the return type of `CustomerSessionElementsSessionManager#fetchCustomerSessionEphemeralKey` from `Result<CachedCustomerEphemeralKey>` to `Result<CachedCustomerEphemeralKey.Available>` (which is what I think it will always return anyways?)

Reply:
This really just moves where the exception is thrown, but I think it's nice for the fetch function to handle it, rather than each caller being responsible for handling the `None` case, when it should never occur

Reply:
Ah that makes more sense actually! Going to follow up with this!